src/sfizz/modulations/ModMatrix.cpp

```diff
@@ -0,0 +1,427 @@
+// SPDX-License-Identifier: BSD-2-Clause
+
+// This code is part of the sfizz library and is licensed under a BSD 2-clause
+// license. You should have receive a LICENSE.md file along with the code.
+// If not, contact the sfizz maintainers at https://github.com/sfztools/sfizz
+
+#include "ModMatrix.h"
+#include "ModId.h"
+#include "ModKey.h"
+#include "ModGenerator.h"
+#include "Buffer.h"
+#include "Config.h"
+#include "SIMDHelpers.h"
+#include "Debug.h"
+#include <absl/container/flat_hash_map.h>
+#include <absl/strings/string_view.h>
+#include <vector>
+#include <algorithm>
+
+namespace sfz {
+
+struct ModMatrix::Impl {
+    double sampleRate_ {};
+    uint32_t samplesPerBlock_ {};
+
+    uint32_t numFrames_ {};
+    NumericId<Voice> voiceId_ {};
```

Comment:
could be named `currentVoiceId_`

Reply:
agreed it's better naming, modified in latest